libs/blocks/global-navigation/utilities/getUserEventHistory.js

```diff
@@ -0,0 +1,31 @@
+/* eslint-disable no-promise-executor-return, no-async-promise-executor */
+import { getConfig } from '../../../utils/utils.js';
+
+let userEventHistory;
+const getUserEventHistory = async (id) => {
+  if (userEventHistory) return userEventHistory;
+  if (!window.adobeIMS?.isSignedInUser()) return new Promise((resolve, reject) => { reject(new Error('User not signed in')); });
+  let resolve;
+  let reject;
+  userEventHistory = userEventHistory || new Promise((_resolve, _reject) => {
+    resolve = _resolve;
+    reject = _reject;
+  });
+  const profile = await window.adobeIMS.getProfile();
+  const guid = profile.userId;
+  const apiUrl = getConfig().env.name === 'prod'
+    ? 'https://www.adobe.com/gating/api/realtime-profile'
+    : 'https://www.stage.adobe.com/gating/api/realtime-profile';
+  const res = await window
+    .fetch(`${apiUrl}?guid=${id || guid}`, {
```

Comment:
Completely trivial nit: you're using `guid` just once, only if an `id` hasn't been passed to the function. If an `id` is available, you're saving the `guid` without it ever being used, so I'd skip the variable assignment and just do `id || profile.userId` here.